feat: update the layout of the profile section on the author page

mamieorine opened the pull request:
Fixes #9841 

The existing author page needs to be updated with the design improvements we have in Figma.


Changes proposed in this pull request:

- Update the social media and accessible links (Homepage, Glitch, Twitter, Github, Linkedin, Mastodon, RSS, and the d.d.c. author's profile) into the profile section.

- Adds an author bio section under the social media list, getting the bio from the bio key (depending on the exitance of a key) in the `src/site/_data/i18n/authors.yml`.

- Adds a fallback string in the case of no i18n result. This is needed because I added the [bio](https://github.com/GoogleChrome/web.dev/pull/9872/files#diff-90d12317ec6c6ca3f3aaf045b4132be3e189606ccd6fc5783fab51fcc937ca86) key into the author collections, It will throw an error if there are some authors in the `src/site/_data/i18n/authors.yml` has no `bio` key. 

- Update the author profile usage in the handbook: 
   https://deploy-preview-9872--web-dev-staging.netlify.app/handbook/author-profile/.

Please review: https://deploy-preview-9872--web-dev-staging.netlify.app/authors/paulkinlan/

matthiasrohmer commented on src/site/content/en/handbook/author-profile/index.md:
```diff
@@ -50,15 +72,18 @@ description: |
+     "ddc": "paulkinlan",
+     "mastodon": "https://status.kinlan.me/@paul",
+     "linkedin": "https://uk.linkedin.com/in/paulkinlan",
+     "image": "image/T4FyVKpzu4WKF1kBNvXepbi08t52/0O1ZGr2P0l9oTKabyUK5.jpeg",
```
The trailing comma would cause a validation error if someone copies this.

matthiasrohmer commented on types/site/_data/authorsData.d.ts:
```diff
@@ -22,6 +22,8 @@ declare global {
     homepage?: string;
     image?: string;
     twitter?: string;
+    ddc?: string;
```
The internal abbreviation we use is `dcc`, this needs to be updated in a bunch of places in this PR.

matthiasrohmer commented on src/site/_filters/i18n.js:
```diff
@@ -64,9 +64,10 @@ const data = {i18n: walk(path.join(__dirname, '..', '_data', 'i18n'))};
  * @return {string}
  */
-const i18n = (path, locale = defaultLocale) => {
+const i18n = (path, locale = defaultLocale, fallback = null) => {
```
What's that needed for?

A problem with this is also that webdev-infra now contains i18n code which is meant to replace the one here.

mamieorine replied:
I added the fallback for no i18n result because I added this `bio: i18n.authors.${key}.bio` to read the translation for the bio key in each author, see [this file](https://github.com/GoogleChrome/web.dev/pull/9872/files#diff-90d12317ec6c6ca3f3aaf045b4132be3e189606ccd6fc5783fab51fcc937ca86).

The current logic is if there is no i18n result, it will throw the error `Could not find i18n result for: ${path}`, and each author in this author.yaml has no bio key yet. So according to the current logic, Is that okay if I return an empty string instead of throwing the error message in the case of no i18n result?

matthiasrohmer reviewed (commented):
Mostly looks good. I am going to ask Paul if he's fine with us using his author profile as demo ground for the new features.

@rachelandrew, for handbook updates and visibility.

rachelandrew commented on src/site/_data/authorsData.json:
```diff
@@ -284,6 +284,11 @@
     "twitter": "paul_kinlan",
+    "github": "PaulKinlan",
+    "glitch": "PaulKinlan",
+    "ddc": "paulkinlan",
```
should that be dcc? Is it referring to developer.chrome.com?

matthiasrohmer commented:
Paul just agreed to to have his profile updated. 🙂

mamieorine commented:
@matthiasrohmer  I have fixed all your requested changes. In the case of no i18n result, I have fixed it by returning an empty string instead of throwing the error. Please let me know what you think 😊

It is ready for re-review: https://deploy-preview-9872--web-dev-staging.netlify.app/authors/paulkinlan/

Thank you!
